Convert IPC::Signal to the new IPC serialization format

cdumez opened the pull request:
#### 4d873d064fe0dbd50803406b2ca406342951cf99
<pre>
Convert IPC::Signal to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=268393">https://bugs.webkit.org/show_bug.cgi?id=268393</a>

Reviewed by Matt Woodrow.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/IPC/IPCEvent.h:
(IPC::Signal::takeSendRight):
(IPC::Signal::semaphore const):
(IPC::Signal::encode): Deleted.
(IPC::Signal::decode): Deleted.
* Source/WebKit/Platform/IPC/IPCEvent.serialization.in: Added.
* Source/WebKit/Platform/IPC/darwin/IPCEventDarwin.cpp:
(IPC::Signal::encode): Deleted.
(IPC::Signal::decode): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

* Tools/TestWebKitAPI/Tests/IPC/EventTests.cpp:
The test was encoding a single Signal object on sender side. However,
on recipient side, it was decoding first a uint64_t and then a Signal.
I have no idea why it was trying to decode that uint64_t given that we
don&apos;t send one. One of the tests started failing on iOS only when
decoding that uint64_t. As a result, I tried dropping the decoding of
this uint64_t in ALL the tests and they are all still passing.

Canonical link: <a href="https://commits.webkit.org/273956@main">https://commits.webkit.org/273956@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0612ce0897c5872f80114d79573caeb502dee89e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37273 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16170 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39564 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39830 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/33242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38577 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18754 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13315 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/31714 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37838 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13637 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32770 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11874 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11861 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33417 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41090 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33741 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33850 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/37780 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12214 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9995 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35926 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/13863 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/8422 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12565 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4833 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/13116 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Tools/TestWebKitAPI/Tests/IPC/EventTests.cpp:
```diff
@@ -83,7 +83,6 @@ TEST_P(EventTestABBA, SerializeAndSignal)
         ASSERT_TRUE(openB());
 
         bClient().setAsyncMessageHandler([&] (IPC::Decoder& decoder) -> bool {
-            decoder.decode<uint64_t>();
```
This decoding was always failing. However, we were not checking if it succeeded or not. Despite failing, we would successfully decode the Signal on the next line with the old decoder.

However, the new generated code does:
```
auto signal = decoder.decode<IPC::Signal>(); // Succeeds and does NOT return nullopt.
if (!decoder.isValid())
    return std::nullopt; // We return here because the decoder was marked as invalid by the earlier decoding of the uint64_t.
```

mattwoodrow replied:
I have no explanations for why it was trying to decode an int before the actual signal.

cdumez committed:
Convert IPC::Signal to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=268393

Reviewed by Matt Woodrow.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/IPC/IPCEvent.h:
(IPC::Signal::takeSendRight):
(IPC::Signal::semaphore const):
(IPC::Signal::encode): Deleted.
(IPC::Signal::decode): Deleted.
* Source/WebKit/Platform/IPC/IPCEvent.serialization.in: Added.
* Source/WebKit/Platform/IPC/darwin/IPCEventDarwin.cpp:
(IPC::Signal::encode): Deleted.
(IPC::Signal::decode): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

* Tools/TestWebKitAPI/Tests/IPC/EventTests.cpp:
The test was encoding a single Signal object on sender side. However,
on recipient side, it was decoding first a uint64_t and then a Signal.
I have no idea why it was trying to decode that uint64_t given that we
don't send one. One of the tests started failing on iOS only when
decoding that uint64_t. As a result, I tried dropping the decoding of
this uint64_t in ALL the tests and they are all still passing.

Canonical link: https://commits.webkit.org/273956@main
